server/src/util/shiftMiddleware.js

```diff
@@ -0,0 +1,56 @@
+import prisma from "../../prisma/client.js";
+import { StatusCodes } from "http-status-codes";
+
+export const isShiftId = async (req, res, next) => {
+  const id = parseInt(req.params.id, 10);
+  const query = await prisma.shift.findUnique({
+    where: {
+      id,
+    },
+  });
+  if (query === null || query === undefined) {
+    return res
+      .status(StatusCodes.CONFLICT)
+      .json({ msg: "ERROR: shift does not exist" });
+  } else {
+    next();
+  }
+};
+
+export const isUserId = async (req, res, next) => {
+  const { id } = req.body;
+  const query = await prisma.user.findUnique({
+    where: {
+      id,
+    },
+  });
+  if (query === null || query === undefined) {
+    return res
+      .status(StatusCodes.CONFLICT)
+      .json({ msg: "ERROR: user does not exist" });
+  } else {
+    next();
+  }
+};
+
+
+export const isValidTimeShift= async (req, res, next) => {
+  const { start, end } = req.body;
+  const query = await prisma.shift.findUnique({
```

Comment:
this query will fail since start and end are not necessarily unique, try findFirst instead

Reply:
be2e55d6fe77dee6c7e9473aaf103d7d2c888fea